amostra/client/amutils.py

```diff
@@ -17,9 +16,10 @@ def doc_or_uid_to_uid(doc_or_uid):
     uid : str
         A string version of the uid of the given document
     """
-    if not isinstance(doc_or_uid, six.string_types):
-        doc_or_uid = doc_or_uid['uid']
-    return str(doc_or_uid)
+    try:
+        return str(doc_or_uid['uid'])
+    except KeyError:
```

Comment:
This should be a TypeError in addition to KeyError bc if uid is a string, it complains string indices must be ints.